### PR TITLE
Docs/operator subpages

### LINKED ISSUE #8084: Breakout Operator API endpoints into their own pages
Nomad's operator API page https://www.nomadproject.io/api-docs/operator/ contains endpoint information for 7 different endpoints. Move these into their own sub page to improve discoverability and readability.

Consul has this for some of their endpoints as well.
image

### PR DESCRIPTION
**Pull request by drewbailey**
resolves #8084

https://nomad-q659y1d2y.vercel.app/api-docs/operator (this is an older commit without updated index title)

### CODE REVIEW

File: website/pages/api-docs/operator/index.mdx [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+sidebar_title: Operator
+description: |-
+  The /operator endpoints provides cluster-level tools for Nomad operators, such
+  as interacting with the Raft subsystem.
+---
+
+# Operator HTTP API
+
+The `/operator` endpoints provide cluster-level tools for Nomad operators, such
+as interacting with the Raft subsystem, licensing, snapshots, autopilot and scheudler configration.
```
Comment (krishicks) [inline website/pages/api-docs/operator/index.mdx:13]:
```suggestion
as interacting with the Raft subsystem, licensing, snapshots, autopilot and scheduler configuration.
```

File: website/pages/api-docs/operator/raft.mdx [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+      "RaftProtocol": 2,
+      "Voter": true
+    }
+  ]
+}
+```
+
+#### Field Reference
+
+- `Index` `(int)` - The `Index` value is the Raft corresponding to this
```
Comment (krishicks) [inline website/pages/api-docs/operator/raft.mdx:63]:
```suggestion
- `Index` `(int)` - The `Index` value is the Raft commit index corresponding to this
```

File: website/pages/api-docs/operator/raft.mdx [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+  - `ID` `(string)` - The ID of the server. This is the same as the `Address`
+    but may be upgraded to a GUID in a future version of Nomad.
+
+  - `Node` `(string)` - The node name of the server, as known to Nomad, or
+    `"(unknown)"` if the node is stale and not known.
+
+  - `Address` `(string)` - The `ip:port` for the server.
+
+  - `Leader` `(bool)` - is either "true" or "false" depending on the server's
```
Comment (krishicks) [inline website/pages/api-docs/operator/raft.mdx:78]:
```suggestion
  - `Leader` `(bool)` - is "true" or "false" depending on the server's
```

File: website/pages/api-docs/operator/raft.mdx [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+
+- `id` `(string: <optional>)` - Specifies the server to remove as
+  `id`. This cannot be provided along with the `address` parameter.
+
+### Sample Request
+
+```shell-session
+$ curl \
+    --request DELETE \
+    https://localhost:4646/v1/operator/raft/peer?address=1.2.3.4
```
Comment (krishicks) [inline website/pages/api-docs/operator/raft.mdx:115]: This example does not include a port; based on the docs above, it's meant to be `ip:port`.

File: website/pages/api-docs/operator/scheduler.mdx [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+  The /operator/scheduler endpoints provide tools for management of Nomad server scheduler settings.
+---
+
+# Scheduler Operator HTTP API
+
+The `/operator/scheduler` endpoints provide tools for management of Nomad server scheduler settings.
+
+## Read Scheduler Configuration
+
+This endpoint retrieves the latest Scheduler configuration. This API was introduced in
```
Comment (krishicks) [inline website/pages/api-docs/operator/scheduler.mdx:15]:
```suggestion
This endpoint retrieves the latest scheduler configuration. This API was introduced in
```

File: website/pages/api-docs/operator/scheduler.mdx [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+    - `SystemSchedulerEnabled` `(bool: true)` - Specifies whether preemption for system jobs is enabled. Note that
+      this defaults to true.
+
+    - `BatchSchedulerEnabled` `(bool: false)` - Specifies whether preemption for batch jobs is enabled. Note that
+      this defaults to false and must be explicitly enabled.
+
+    - `ServiceSchedulerEnabled` `(bool: false)` - Specifies whether preemption for service jobs is enabled. Note that
+      this defaults to false and must be explicitly enabled.
+
+  - `CreateIndex` - The Raft index at which the config was created.
```
Comment (krishicks) [inline website/pages/api-docs/operator/scheduler.mdx:79]:
```suggestion
  - `CreateIndex` - The Raft commit index at which the config was created.
```

File: website/pages/api-docs/operator/scheduler.mdx [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+  - `ModifyIndex` - The Raft index at which the config was modified.
```
Comment (krishicks) [inline website/pages/api-docs/operator/scheduler.mdx:80]:
```suggestion
  - `ModifyIndex` - The Raft commit index at which the config was modified.
```

File: website/pages/api-docs/operator/scheduler.mdx [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+
+### Bootstrap Configuration Element
+
+The [`default_scheduler_config`][] attribute of the server stanza will provide a
+starting value for this configuration. Once bootstrapped, the value in the
+server state is authoritative.
+
+### Parameters
+
+- `cas` `(int: 0)` - Specifies to use a Check-And-Set operation. The update will
```
Comment (krishicks) [inline website/pages/api-docs/operator/scheduler.mdx:106]:
```suggestion
- `cas` `(int: 0)` - Specifies to use a check-and-set operation. The update will
```

File: website/pages/api-docs/operator/scheduler.mdx [hunk 9]
```diff
@@ -0,0 +1,157 @@
...
+    "SystemSchedulerEnabled": true,
+    "BatchSchedulerEnabled": false,
+    "ServiceSchedulerEnabled": true
+  }
+}
+```
+
+- `SchedulerAlgorithm` `(string: "binpack")` - Specifies whether scheduler
+  binpacks or spreads allocations on available nodes. Possible values are
+  `"binpack"` and `"spread"`
```
Comment (krishicks) [inline website/pages/api-docs/operator/scheduler.mdx:125]:
```suggestion
  `"binpack"` and `"spread"`.
```

File: website/pages/api-docs/operator/scheduler.mdx [hunk 10]
```diff
@@ -0,0 +1,157 @@
...
+- `PreemptionConfig` `(PreemptionConfig)` - Options to enable preemption for
+  various schedulers.
+
+  - `SystemSchedulerEnabled` `(bool: true)` - Specifies whether preemption for
+    system jobs is enabled. Note that if this is set to true, then system jobs
+    can preempt any other jobs.
+
+  - `BatchSchedulerEnabled` `(bool: false)` - Specifies
+    whether preemption for batch jobs is enabled. Note that if this is set to
+    true, then batch jobs can preempt any other jobs.
```
Comment (krishicks) [inline website/pages/api-docs/operator/scheduler.mdx:136]:
```suggestion
    true, batch jobs can preempt any other jobs.
```

File: website/pages/api-docs/operator/scheduler.mdx [hunk 11]
```diff
@@ -0,0 +1,157 @@
...
+
+  - `ServiceSchedulerEnabled` `(bool: false)` - Specifies
+    whether preemption for service jobs is enabled. Note that if this is set to
+    true, then service jobs can preempt any other jobs.
```
Comment (krishicks) [inline website/pages/api-docs/operator/scheduler.mdx:140]:
```suggestion
    true, service jobs can preempt any other jobs.
```

File: website/pages/api-docs/operator/snapshot.mdx [hunk 12]
```diff
@@ -0,0 +1,87 @@
...
+
+## Generate Snapshot
+
+This endpoint generates and returns an atomic, point-in-time snapshot of the
+Nomad server state for disaster recovery. Snapshots include all state managed by Nomad's
+Raft [consensus protocol](/docs/internals/consensus).
+
+Snapshots are exposed as gzipped tar archives which internally contain the Raft
+metadata required to restore, as well as a binary serialized version of the
+Nomad server state. The contents are covered internally by SHA-256 hashes.
```
Comment (krishicks) [inline website/pages/api-docs/operator/snapshot.mdx:19]: Covered? Maybe, verified?

**Review by krishicks — CHANGES_REQUESTED** [review]

File: website/pages/api-docs/operator/scheduler.mdx (hunk 8 above)
Comment (drewbailey) [reply website/pages/api-docs/operator/scheduler.mdx:106]: The docs seem to reference Check-And-Set elsewhere so hesitant to change, all of these docs changes where cut/paste

File: website/pages/api-docs/operator/scheduler.mdx (hunk 7 above)
Comment (drewbailey) [reply website/pages/api-docs/operator/scheduler.mdx:80]: Same as some of the otheres, but Raft index seems to be used widely in the docs / internal comments but raft commit index is not, hesitant to change

File: website/pages/api-docs/operator/index.mdx (hunk 1 above)
Comment (drewbailey) [reply website/pages/api-docs/operator/index.mdx:13]: woops!

File: website/pages/api-docs/operator/snapshot.mdx (hunk 12 above)
Comment (drewbailey) [reply website/pages/api-docs/operator/snapshot.mdx:19]: :man_shrugging: copy-paste, Verified does seem a bit clearer

Comment (drewbailey) [inline website/pages/api-docs/operator/snapshot.mdx:19]:
```suggestion
Nomad server state. The contents are verified internally by SHA-256 hashes.
```

**Review by mazn403 — APPROVED** [review]

File: website/pages/api-docs/operator/autopilot.mdx [hunk 13]
```diff
@@ -0,0 +1,214 @@
...
+layout: api
+page_title: Autopilot - Operator - HTTP API
+sidebar_title: Autopilot
+description: |-
+  The /operator/autopilot endpoints provide tools for managing Autopilot.
+---
+
+# Autopilot Operator HTTP API
+
+The `/operator/autopilot` endpoints allow for automatic operator-friendly management of Nomad servers including cleanup of dead servers, monitoring the state of the Raft cluster, and stable server introduction.
```
Comment (tgross) [inline website/pages/api-docs/operator/autopilot.mdx:11]: We don't have a linter for it yet (although the docs folks say one is coming), but most of our docs hard-wrap markdown lines.

Comment (tgross) [inline website/pages/api-docs/operator/autopilot.mdx:11]: We don't have a linter for it yet (although the docs folks say one is coming), but most of our docs hard-wrap markdown lines.

**Review by tgross — APPROVED** [review]
Overall this is a solid improvement of the page flow 👍

File: website/pages/api-docs/operator/scheduler.mdx (hunk 7 above)
Comment (krishicks) [reply website/pages/api-docs/operator/scheduler.mdx:80]: I noticed inconsistencies within this doc; sometimes it's Raft commit index, sometimes just index. I'm fine with either, but consistency is nice.

File: website/pages/api-docs/operator/raft.mdx (hunk 4 above)
Comment (drewbailey) [reply website/pages/api-docs/operator/raft.mdx:115]: :+1: added

File: website/pages/api-docs/operator/scheduler.mdx (hunk 7 above)
Comment (drewbailey) [reply website/pages/api-docs/operator/scheduler.mdx:80]: :+1: converging on `Raft index`

**Review by krishicks — APPROVED** [review]
:tada:

**Pull request merged**